### PR TITLE
Adjust System to support the new generate_ode_function. Fixes #116

### LINKED ISSUE #116: Adjust System to support the new generate_ode_function.
System is setup to use the now deprecated function `generate_ode_function` from the `pydy.codegen.code` module. It needs to be changed to use `pydy.codegen.ode_function_generators.generate_ode_function`.

### PR DESCRIPTION
**Pull request by jinankjain**
- [ ] There are no merge conflicts.
- [ ] If there is a related issue, a reference to that issue is in the
  commit message.
- [ ] Unit tests have been added for the bug. (Please reference the issue #
  in the unit test.)
- [ ] The tests pass both locally (run `nosetests`) and on Travis CI.
- [ ] The code follows PEP8 guidelines. (use a linter, e.g.
  [pylint](http://www.pylint.org), to check your code)
- [ ] The new feature is documented in the [Release
  Notes](https://github.com/pydy/pydy/tree/contributing#release-notes).
- [ ] The code is backwards compatible. (All public methods/classes must
  follow deprecation cycles.)
- [ ] All reviewer comments have been addressed.

### CODE REVIEW

File: pydy/system.py [hunk 1]
```diff
@@ -462,18 +463,37 @@ def generate_ode_function(self, generator='lambdify', **kwargs):
 
         """
 
-        if 'specified' in kwargs:
-            kwargs.pop('specified')
-            print("User supplied 'specified' kwarg was disregarded.")
+        try:
```
Comment (tarzzz) [inline pydy/system.py]: Why are you putting everything in a `try-except` block? You can simply check if "Specified" is in kwargs, then issue a warning, and move ahead.

Comment (jinankjain) [inline pydy/system.py]: As Jason mentioned he wants support of old library still so I used to try-except block

Comment (moorepants) [inline pydy/system.py]: Yes, please make this backwards compatible. There is a lot of code in the wild that would like to get a warning about this minor change.

Comment (jinankjain) [inline pydy/system.py]: Yes, this one supports both versions as I have imported both function.

**Comment by moorepants** [comment]
I just read the code and realized that all this is not necessary. System handles `specifieds` internally. So I think all that needs to happen is to change that. The user doesn't ever type `specifieds` and if they do pass it to the `kwargs` it gets popped out. Deprecation is unlikely needed.

**Comment by moorepants** [comment]
This function needs to be changed to use `specifieds`: `_kwargs_for_gen_ode_func()`

**Comment by moorepants** [comment]
And `generate_ode_function` needs to pop either `specifieds` or `specified` because the user could pass in either.

**Comment by jinankjain** [comment]
I have incorporated all the changes could you please review it.

**Comment by moorepants** [comment]
@jinankjain I think it would help if you wrote a test first. For example, `System()` should be able to accept `specifieds` or `specified` as a key word arg.

**Comment by moorepants** [comment]
Also, shouldn't we always be using the new `generate_ode_function`? Why use the old one at all? Isn't the whole purpose to remove the old version?

**Comment by moorepants** [comment]
The new `generate_ode_function` only accepts `specifieds`.

You need to write a unit test for this first. If you write a unit test(s) for the `__init__` method then you can run the tests locally to see if they pass.

**Comment by moorepants** [comment]
@jinankjain I've added a checklist to the top of the PR. Once you address those issues and a reviewer has checked them off, we can merge.

**Comment by moorepants** [comment]
PR #154 fixes the issue. Closing this one.

**Pull request closed without merging**